agent/health_endpoint.go

```diff
@@ -186,7 +186,7 @@ func (s *HTTPServer) healthServiceNodes(resp http.ResponseWriter, req *http.Requ
 	var out structs.IndexedCheckServiceNodes
 	defer setMeta(resp, &out.QueryMeta)
 
-	if args.QueryOptions.UseCache || args.MinQueryIndex > 0 {
+	if (args.QueryOptions.UseCache || args.MinQueryIndex > 0) && !connect {
```

Comment:
Why only non-connect?

Reply:
When `connect = true` we get effectively a different RPC endpoint (different state store logic/table lookup) so I think it'll require a different topic. The current endpoint shares filtering logic since the return types are the same but swaps the state store lookup function based on whether or not connect is enabled.

Reply:
🤔 It would be _really_ nice to support connect at launch since that is one of the biggest use-cases for discovery in our service mesh model.

I think you're right about separate topic though so it seems OK to do that in a separate PR but maybe add a TODO to the project board?